services/core-api/app/api/users/minespace/resources/minespace_user.py

```diff
@@ -63,3 +68,36 @@ def delete(self, user_id):
         db.session.delete(user)
         db.session.commit()
         return ('', 204)
+
+    @api.doc(description='Update an existing Minespace Users mine list')
+    @api.marshal_with(MINESPACE_USER_MODEL, envelope='records')
+    @requires_role_mine_admin
+    def put(self, user_id):
+        contact = MinespaceUser.find_by_id(user_id)
+        if not contact:
+            raise NotFound('Contact not found.')
+        data = self.parser.parse_args()
+        if not data.get('mine_guids'):
+            raise BadRequest('Empty list mine_guids is not permitted. Please provide a list of mine GUIDS.')
+
+        existing_mines = contact.mines # list of mines already existing in the user's mine list
+        updated_mines = data.get('mine_guids') # updated list of mines to be applied to the user
+
+        for delete_mine in existing_mines:
+            if str(delete_mine) not in updated_mines:
+                minespace_user_mine = MinespaceUserMine.find_by_minespace_user_mine_relationship(delete_mine, user_id)
+                if minespace_user_mine:  
+                    minespace_user_mine.delete()
+
+        # Cycle through list of mines. Mines have to exist before being added to the user.
+        for guid in updated_mines:
+            mine = Mine.find_by_mine_guid(guid)
+            if not mine:
+                raise NotFound('Mine with guid {} not found.'.format(guid))
+            existing_minespace_user_mine = MinespaceUserMine.find_by_minespace_user_mine_relationship(guid, user_id)
+            current_app.logger.debug('Existing Mine: {}'.format(existing_minespace_user_mine))
+            if not existing_minespace_user_mine:
+                new_minespace_user_mine = MinespaceUserMine.create(user_id, mine.mine_guid)       
+        contact.save()
+        return MinespaceUser.get_all()
```

Comment:
this put is updating a specific user and its associated mines so I am not sure why the return statement is all minespaceusers.

Reply:
Fixed.